pkg/config/types/jobdefaults.go

```diff
@@ -46,6 +46,28 @@ type ResourcesConfig struct {
 	GPU string `yaml:"GPU,omitempty"`
 }
 
+type DefaultPublisherConfig struct {
+	// Type specifies the publisher type. e.g. "s3", "local", "ipfs", etc.
+	Type string `json:"Type" yaml:"Type,omitempty"`
+
+	// Params specifies the publisher configuration data.
+	Params map[string]string `json:"Params,omitempty" yaml:"Params,omitempty"`
```

Comment:
why params values string? We should enable integers and booleans as well, right?

Reply:
Explained out of band here, In short: we lack a schema to know how to parse values we receive from the config to their required types. We've decided this is acceptable for now.